src/StreamJsonRpc.Tests/MessagePackFormatterTests.cs

```diff
@@ -39,6 +39,59 @@ public void JsonRpcRequest_ArgsArray()
         Assert.Equal(((CustomType)original.ArgumentsList[2]).Age, ((CustomType)actual.ArgumentsList[2]).Age);
     }
 
+    [Fact]
+    public void JsonRpcRequest_ProgressArg()
+    {
+        var original = new JsonRpcRequest
+        {
+            Id = 5,
+            Method = "test",
+            ArgumentsList = new object[] { new Progress<object>() },
+        };
+
+        var actual = this.Roundtrip(original);
+        Assert.Equal(original.Id, actual.Id);
+        Assert.Equal(original.Method, actual.Method);
+        Assert.Equal(typeof(long), actual.ArgumentsList[0].GetType());
+    }
+
+    [Fact]
+    public void JsonRpcRequest_MultipleProgressArgs()
+    {
+        var original = new JsonRpcRequest
+        {
+            Id = 5,
+            Method = "test",
+            ArgumentsList = new object[] { new Progress<object>(), new Progress<object>(), new Progress<object>() },
+        };
+
+        var actual = this.Roundtrip(original);
+        Assert.Equal(original.Id, actual.Id);
+        Assert.Equal(original.Method, actual.Method);
+        Assert.Equal(typeof(long), actual.ArgumentsList[0].GetType());
+        Assert.Equal(typeof(long), actual.ArgumentsList[1].GetType());
+        Assert.Equal(typeof(long), actual.ArgumentsList[2].GetType());
+    }
+
+    [Fact]
+    public void JsonRpcRequest_MultipleArgsIncludingProgress()
+    {
+        var original = new JsonRpcRequest
+        {
+            Id = 5,
+            Method = "test",
+            ArgumentsList = new object[] { 0, new Progress<object>(), "foo", new ProgressWithCompletion<object>(report => { }) },
+        };
+
+        var actual = this.Roundtrip(original);
+        Assert.Equal(original.Id, actual.Id);
+        Assert.Equal(original.Method, actual.Method);
+        Assert.Equal(original.ArgumentsList[0], actual.ArgumentsList[0]);
+        Assert.Equal(typeof(long), actual.ArgumentsList[1].GetType());
+        Assert.Equal(original.ArgumentsList[2], actual.ArgumentsList[2]);
+        Assert.Equal(typeof(long), actual.ArgumentsList[3].GetType());
+    }
+
     [Fact]
```

Comment:
I want tests to verify that MessagePack works, but it's not clear to me that these tests verify anything interesting other than an implementation detail which represents just a small part of the overall feature.

Reply:
I was not sure how to implement this test, should I add a round trip for a progress notification as well? And maybe for a JsonResponse?

---
In reply to: [303664893](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r303664893) [](ancestors = 303664893)

Reply:
Well, given this is a formatter-specific test class, I suppose use of JsonRpc itself is inappropriate here. We do want to make sure that the JsonRpcTests are exercising the MessagePackFormatter and you've already done that. So at this point, I'm not sure these tests are worthwhile since the same functionality is tested at a higher level.

---
In reply to: [304098095](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r304098095) [](ancestors = 304098095,303664893)